src/main.js

```diff
@@ -223,6 +224,18 @@ const CodingManagerApplication = new Lang.Class({
         });
 
         this._window.connect('notify::visible', Lang.bind(this, this._on_visibility_changed));
+        this._window.connect('map-event', Lang.bind(this, function() {
```

Comment:
I must admit that I preferred my solution. :P I mean - we wanted to listen to the "active window changed" signal when the window is mapped (got the surface from the window manager/display server). Somehow the window state changes happen around the same time too, so it could be used, but it's a bit cryptic.

Anyway, changed it to use it, seems to work too.